### ISSUE TITLE
utf8cpp 4.0.1 has no working imported CMake target

### ISSUE THREAD

**Issue by ufleisch** [Homebrew/homebrew-core#153289]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 4.1.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: bf8991f0d777fe8e33105f0cd7bf1e5993eed60a
Last commit: 6 days ago
Core tap JSON: 03 Nov 20:59 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: quad-core 64-bit penryn
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.5.0.0.1.1617976050
Xcode: N/A
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

How to reproduce:

Try to build https://github.com/taglib/taglib/commits/master [8c63c877], it will fail because utf8.h is not found.
Try to build the following commit [70b4ce79], it will work because the following patch has been added:
diff --git a/taglib/CMakeLists.txt b/taglib/CMakeLists.txt
index 911ce058..89f527d1 100644
--- a/taglib/CMakeLists.txt
+++ b/taglib/CMakeLists.txt
@@ -343,6 +343,16 @@ add_library(tag ${tag_LIB_SRCS} ${tag_HDRS})
 target_link_libraries(tag $<$<TARGET_EXISTS:utf8::cpp>:utf8::cpp>)
 target_link_libraries(tag $<$<TARGET_EXISTS:ZLIB::ZLIB>:ZLIB::ZLIB>)
 
+# Homebrew utf8cpp 4.0.1 seems to have issues. utf8.h is not found because
+# imported target utf8::cpp does not exist. The imported target utf8cpp::utf8cpp
+# includes non-existent path /usr/local/include/utf8cpp and cannot be used.
+# Therefore the following (temporary) hack is used to find the utf8cpp includes.
+if(APPLE)
+  if(EXISTS /usr/local/include/utf8.h)
+    include_directories(/usr/local/include)
+  endif()
+endif()
+
 set_target_properties(tag PROPERTIES
   VERSION ${TAGLIB_SOVERSION_MAJOR}.${TAGLIB_SOVERSION_MINOR}.${TAGLIB_SOVERSION_PATCH}
   SOVERSION ${TAGLIB_SOVERSION_MAJOR}
The comment describes the problem: Target utf8::cpp does not seem to exist. The imported target utf8cpp::utf8cpp from /usr/local/share/utf8cpp/cmake/utf8cppTargets.cmake contains a non existing include path /usr/local/include/utf8cpp and the configure step therefore will fail. Could you please add an imported target utf8::cpp with correct include path?

### What happened (include all command output)?

```
/Users/pcwiz/projects/taglib/src/taglib/taglib/toolkit/tstring.cpp:30:10: fatal error: 'utf8.h' file not found
#include <utf8.h>
```

### What did you expect to happen?

Build to succeed.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install cppunit utf8cpp
git clone https://github.com/taglib/taglib/commits/master
mkdir build
cd taglib
git checkout 8c63c877
cmake -B../build -DBUILD_SHARED_LIBS=ON -DVISIBILITY_HIDDEN=ON -DBUILD_TESTING=ON \
-DBUILD_EXAMPLES=ON -DBUILD_BINDINGS=ON -DCMAKE_BUILD_TYPE=Release
cmake --build ../build --config Release
```

**Comment by SMillerDev** [comment]
Does this also happen if you build it from source?

**Comment by ufleisch** [comment]
Thanks for the fast response.
You can easily build TagLib with utf8cpp from source by using it as a Git submodule. In the Step-by-step reproduction, inside the checked-out taglib folder, type

    git submodule update --init
    brew uninstall utf8cpp
    
(you can afterwards reset it again using `git submodule deinit --all; brew install utf8cpp`). Then reconfigure using the given `cmake` commands, it will now print `Using utfcpp from (..)/taglib/src/taglib/3rdparty/utfcpp` instead of `Using utfcpp 4.0.1 from /usr/local/share/utf8cpp/cmake/utf8cppConfig.cmake`.

    cmake -B../build -DBUILD_SHARED_LIBS=ON -DVISIBILITY_HIDDEN=ON -DBUILD_TESTING=ON \
    -DBUILD_EXAMPLES=ON -DBUILD_BINDINGS=ON -DCMAKE_BUILD_TYPE=Release

    cmake --build ../build --config Release

Now utf8cpp is included as a subdirectory and not as a package. The corresponding part in _taglib/CMakeLists.txt_ is

```
find_package(utf8cpp QUIET)
if(NOT utf8cpp_FOUND)
  if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/utfcpp/CMakeLists.txt)
    add_subdirectory("3rdparty/utfcpp")
    message(STATUS "Using utfcpp from ${utf8cpp_SOURCE_DIR}")
  else()
    message(FATAL_ERROR
      "utfcpp not found. Either install package (probably utfcpp, utf8cpp, or libutfcpp-dev) "
      "or fetch the git submodule using\n"
      "git submodule update --init")
  endif()
else()
  message(STATUS "Using utfcpp ${utf8cpp_VERSION} from ${utf8cpp_CONFIG}")
endif()
```

The library is added as `add_library(utf8::cpp ALIAS utf8cpp)` _taglib/3rdparty/utfcpp/CMakeLists.txt_, that's why it would be good to use the same name `utf8:cpp` but the correct include directory.

**Comment by SMillerDev** [comment]
Taglib isn't a homebrew formula though so that's not really relevant here. What I am trying to determine is if utf8cpp 4.0.1 works everywhere else with CMake, or if the release is simply broken.

**Comment by ufleisch** [comment]
I also think that the release is simply broken and that it is not caused by Homebrew. The bug probably gets easily masked if another dependency adds the same include directory or if utf8cpp is installed into the standard include directory (as it is done in Arch Linux, which also provides the latest 4.0.1 release).
I have analyzed the problem in https://github.com/nemtrif/utfcpp/issues/112#issuecomment-1793455798 and provided a PR to fix it in https://github.com/nemtrif/utfcpp/pull/113.

**Comment by iMichka** [comment]
4.0.2 includes the fix: #153333

**Comment by ufleisch** [comment]
Thanks for the quick response and the new release, it works now.
